platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -6711,42 +6763,9 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 		return;
 	}
 
-	xim = XOpenIM(x11_display, nullptr, nullptr, nullptr);
-
-	if (xim == nullptr) {
-		WARN_PRINT("XOpenIM failed");
-		xim_style = 0L;
-	} else {
-		::XIMCallback im_destroy_callback;
-		im_destroy_callback.client_data = (::XPointer)(this);
-		im_destroy_callback.callback = (::XIMProc)(_xim_destroy_callback);
-		if (XSetIMValues(xim, XNDestroyCallback, &im_destroy_callback,
-					nullptr) != nullptr) {
-			WARN_PRINT("Error setting XIM destroy callback");
-		}
-
-		::XIMStyles *xim_styles = nullptr;
-		xim_style = 0L;
-		char *imvalret = XGetIMValues(xim, XNQueryInputStyle, &xim_styles, nullptr);
-		if (imvalret != nullptr || xim_styles == nullptr) {
-			fprintf(stderr, "Input method doesn't support any styles\n");
-		}
-
-		if (xim_styles) {
-			xim_style = 0L;
-			for (int i = 0; i < xim_styles->count_styles; i++) {
-				const ::XIMStyle &style = xim_styles->supported_styles[i];
-
-				if (!_is_xim_style_supported(style)) {
-					continue;
-				}
-
-				xim_style = _get_best_xim_style(xim_style, style);
-			}
-
-			XFree(xim_styles);
-		}
-		XFree(imvalret);
+	if (!XRegisterIMInstantiateCallback(x11_display, nullptr, nullptr, nullptr, (::XIDProc)(_xim_instantiate_callback), (::XPointer)this)) {
+		WARN_PRINT("Error register XIM instantiate callback");
```

Comment:
Normally we'd end warnings with a period but none of the other warnings in this code seem to follow that convention, so I think it's fine to keep it consistent by not having a period (or we could add one to all warnings here).
```suggestion
		WARN_PRINT("Error registering XIM instantiate callback");
```

Reply:
Thanks for the suggestion, added a period to all warnings.